File: meditime_frontend/lib/features/home/user/rdv/pages/models/doctor_slot_model.dart
class DoctorSlot {
  final int? id;
  final int doctorId;
  final String startDay;
  final int startHour;
  final int startMinute;
  final String endDay;
  final int endHour;
  final int endMinute;
  final String? status; // <-- Ajoute ce champ
  final DateTime? createdAt;

  const DoctorSlot({
    this.id,
    required this.doctorId,
    required this.startDay,
    required this.startHour,
    required this.startMinute,
    required this.endDay,
    required this.endHour,
    required this.endMinute,
    this.status, // <-- Ajoute ce champ
    this.createdAt,
  });

  Map<String, dynamic> toJson() => {
    if (id != null) 'id': id,
    'doctorId': doctorId,
    'startDay': startDay,
    'startHour': startHour,
    'startMinute': startMinute,
    'endDay': endDay,
    'endHour': endHour,
    'endMinute': endMinute,
    if (status != null) 'status': status, // <-- Ajoute ce champ
  };

  factory DoctorSlot.fromJson(Map<String, dynamic> json) => DoctorSlot(
    id: json['id'],
    doctorId: json['doctorId'],
    startDay: json['startDay'],
    startHour: json['startHour'],
    startMinute: json['startMinute'],
    endDay: json['endDay'],
    endHour: json['endHour'],
    endMinute: json['endMinute'],
    status: json['status'], // <-- Ajoute ce champ
    createdAt: json['created_at'] != null 
        ? DateTime.parse(json['created_at'])
        : null,
  );
}

File: meditime-backend/src/rdv/models/doctor_slot_model.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../../config/db');
const Doctor = require('../../models/doctor_model');

const DoctorSlot = sequelize.define('DoctorSlot', {
  id: { type: DataTypes.INTEGER, autoIncrement: true, primaryKey: true },
  doctorId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: { model: 'doctor', key: 'id' }
  },
  startDay: { type: DataTypes.STRING(20), allowNull: false },
  startHour: { type: DataTypes.INTEGER, allowNull: false },
  startMinute: { type: DataTypes.INTEGER, allowNull: false },
  endDay: { type: DataTypes.STRING(20), allowNull: false },
  endHour: { type: DataTypes.INTEGER, allowNull: false },
  endMinute: { type: DataTypes.INTEGER, allowNull: false },
  created_at: { type: DataTypes.DATE, defaultValue: DataTypes.NOW },
  status: { type: DataTypes.ENUM('active', 'expired'), allowNull: false, defaultValue: 'active' }
}, {
  tableName: 'doctor_slots',
  timestamps: false,
  indexes: [
    {
      unique: true,
      fields: ['doctorId', 'startDay', 'startHour', 'startMinute', 'endDay', 'endHour', 'endMinute']
    }
  ]
});

// Association
Doctor.hasMany(DoctorSlot, { foreignKey: 'doctorId', as: 'slots' });
DoctorSlot.belongsTo(Doctor, { foreignKey: 'doctorId', as: 'doctor' });

module.exports = DoctorSlot;